src/backend/distributed/executor/local_executor.c

```diff
@@ -839,7 +843,7 @@ TaskAccessesLocalNode(Task *task)
 void
 ErrorIfTransactionAccessedPlacementsLocally(void)
 {
-	if (GetCurrentLocalExecutionStatus() == LOCAL_EXECUTION_REQUIRED)
+	if (GetCurrentLocalExecutionStatus() >= LOCAL_EXECUTION_REQUIRED_READONLY)
```

Comment:
The naming of LOCAL_EXECUTION_REQUIRED_ seems a bit strange. It's not very intuitive what REQUIRED_MODIFY means and that one status is greater than the other. 

Could this be a separate flag, or are there some better names?

Reply:
Agreed both (a) the naming was not great (b) we should not have an ordering between the status.

I initially tried having one more state, but that seemed more complicated. Now, I pushed a WIP commit that improves (a) and (b). 

Does that look or do you think we should introduce a new state?